### PR TITLE
feat: adding additional metadata fields

### LINKED ISSUE #1: Missing Metadata.Subtype
Found in an event that is thrown when someone plays a trailer on the plex server. Here's an example json:
```
{
  "Account": {
    "id": 1, 
    "thumb": "https://plex.tv/users/123456789a1/avatar?c=123456789", 
    "title": "russorat"
  }, 
  "Metadata": {
    "addedAt": 1613901884, 
    "art": "/library/metadata/1234/art/123456789", 
    "extraType": 1, 
    "guid": "iva://api.internetvideoarchive.com/2.0/DataService/VideoAssets(1234567)?lang=en&bitrates=80,212,450,750,1500,2500,5000,8000&duration=130&adaptive=1&dts=0", 
    "index": 1, 
    "key": "/library/metadata/12345", 
    "originallyAvailableAt": "2019-12-19", 
    "ratingKey": "1234", 
    "subtype": "trailer", 
    "summary": "", 
    "thumb": "/library/metadata/1234/thumb/123456", 
    "title": "My Cool Movie", 
    "type": "clip", 
    "year": 2019
  }, 
  "Player": {
    "local": true, 
    "publicAddress": "11.11.11.11", 
    "title": "Chrome", 
    "uuid": "xxxxxxxxxxxx"
  }, 
  "Server": {
    "title": "ExampleServer", 
    "uuid": "xxxxxxxxxxxxx"
  }, 
  "event": "media.play", 
  "owner": true, 
  "user": true
}
```

### PR DESCRIPTION
**Pull request by russorat**
thanks for this project!

I added some missing metadata fields that were missing, and sorted the list of metadata fields to make it easier to find others that are missing. 

closes #1

### CODE REVIEW

**Comment by russorat** [comment]
added GenuineMediaAnalysis and Live as well which i saw when playing media from live tv

```
{
  "Account": {
    "id": 1, 
    "thumb": "https://plex.tv/users/xxxxxxxx/avatar?c=xxxxxxxx", 
    "title": "xxxxxxxx"
  }, 
  "Metadata": {
    "addedAt": 1614207650, 
    "genuineMediaAnalysis": "1", 
    "grandparentThumb": "https://metadata-static.plex.tv/0/gracenote/xxxxxxxxxxx.jpg", 
    "grandparentTitle": "Match Game", 
    "guid": "plex://episode/xxxxxxxxxxx", 
    "key": "/livetv/sessions/6744d5bc-f1f8-4742-8bb5-xxxxxxxxxxx", 
    "live": "1", 
    "parentIndex": 2021, 
    "ratingKey": "8368", 
    "summary": "", 
    "title": "Match Game", 
    "type": "episode", 
    "year": 2021
  }, 
  "Player": {
    "local": true, 
    "publicAddress": "xx.xx.xx.xx", 
    "title": "Chrome", 
    "uuid": "xxxxxxxxxxx"
  }, 
  "Server": {
    "title": "xxxxxx", 
    "uuid": "xxxxxxxxx"
  }, 
  "event": "media.play", 
  "owner": true, 
  "user": true
}
```

**Comment by russorat** [comment]
few more metadata fields from playing a streaming movie:

```
{
  "Account": {
    "id": 12345, 
    "title": "xxxxxxx"
  }, 
  "Metadata": {
    "addedAt": 1612166400, 
    "art": "https://image.tmdb.org/t/p/original/eJ6kPtua2rEvjxHdCtnIobXJZSf.jpg", 
    "attributionLogo": "https://provider-static.plex.tv/vod/partners/logos/crackle.png", 
    "audienceRating": 9.3, 
    "audienceRatingImage": "rottentomatoes://image.rating.upright", 
    "banner": "https://assets.fanart.tv/fanart/movies/829/moviebanner/chinatown-5318abafa0ef1.jpg", 
    "contentRating": "R", 
    "duration": 7829994, 
    "guid": "plex://movie/xxxxxxxx", 
    "indirect": true, 
    "key": "/library/metadata/xxxxxxxx", 
    "originallyAvailableAt": "1974-06-20T00:00:00Z", 
    "primaryExtraKey": "/library/metadata/6015de7207408f002dcee048/extras/iva-65", 
    "rating": 9.9, 
    "ratingCount": 1933, 
    "ratingImage": "rottentomatoes://image.rating.ripe", 
    "ratingKey": "6015de7207408f002dcee048", 
    "studio": "Paramount", 
    "thumb": "https://image.tmdb.org/t/p/original/7ljxmU9L0ugUvyyg0GBkFaarjEq.jpg", 
    "title": "Chinatown", 
    "type": "movie", 
    "viewCount": 0, 
    "viewOffset": 0, 
    "year": 1974
  }, 
  "Player": {
    "local": false, 
    "publicAddress": "xx.xx.xx.xx", 
    "title": "Plex Web (Chrome)", 
    "uuid": "xxxxxxx"
  }, 
  "Server": {
    "title": "Vod", 
    "uuid": "tv.plex.provider.vod"
  }, 
  "event": "media.stop", 
  "owner": false, 
  "user": true
}
```

**Comment by hekmon** [comment]
LGTM !

File: metadata.go [hunk 1]
```diff
@@ -15,8 +15,10 @@ import (
 type Metadata struct {
 	AddedAt               time.Time          `json:"addedAt"`               // movie + show + music
 	Art                   string             `json:"art"`                   // movie + show
+	AttributionLogo       string             `json:"attributionLogo"`       // streaming movie
 	AudienceRating        float64            `json:"audienceRating"`        // movie
 	AudienceRatingImage   string             `json:"audienceRatingImage"`   // movie
+	Banner                string             `json:"banner"`                // movie
```
Comment (hekmon) [inline metadata.go:21]: Just curious... what value can this hold ? (I am thinking URL)

Comment (hekmon) [reply metadata.go:21]: just saw ur payloads on the update PR, I will update them

**Comment by russorat** [comment]
yeah i also saw some different jsons for photos as well. i'll open a new pr with those updates if you're interested.

**Comment by hekmon** [comment]
I just [converted](https://github.com/hekmon/plexwebhooks/compare/f8722b2b0b563743dae17fb8258ebe7b4a0c62e9...515ee4711f15872a2be8f64be687cccb926cd3b5) `AttributionLogo` and `Banner` to optionnal URL (among other things).

Leaving:
* ViewOffset: could it be converted to time.Duration in your opinion ? not sure what is is... (seconds ?)
* GenuineMediaAnalysis: value is string with number "1" on your example... sounds like a boolean to me, should we convert it ?
* Live: same

If you have more payloads/properties plz send a PR ! I would be more than happy to make this lib as complete as possible :)

I did not see tha others were using it so once we have sorted everything out... I hope to tag it v1.0.0 now than some else than me tested it !

**Comment by hekmon** [comment]
Also, during dev don't hesite to uncomment [this line](https://github.com/hekmon/plexwebhooks/blob/v0.1.0/extract.go#L37) in order to quickly find new properties on you payloads :)

**Pull request merged**